mlflow/utils/proto_json_utils.py

```diff
@@ -1,17 +1,79 @@
 import base64
 import datetime
 
+import json
 from json import JSONEncoder
 
 from google.protobuf.json_format import MessageToJson, ParseDict
+from google.protobuf.descriptor import FieldDescriptor
 
 from mlflow.exceptions import MlflowException
 from collections import defaultdict
+from functools import partial
+
+
+def _mark_int64_fields(proto_message):
+    """Converts a proto message to JSON, preserving only the int64/uint64/fixed64 fields."""
+
+    json_dict = {}
+    for field, value in proto_message.ListFields():
+        if field.type == FieldDescriptor.TYPE_MESSAGE:
+            ftype = partial(_mark_int64_fields)
+        elif field.type in [
+            FieldDescriptor.TYPE_INT64,
+            FieldDescriptor.TYPE_UINT64,
+            FieldDescriptor.TYPE_FIXED64,
+        ]:
+            ftype = int
+        else:
+            # Skip all non-int64 fields.
+            continue
+
+        if field.label == FieldDescriptor.LABEL_REPEATED:
+            json_value = []
+            for v in value:
+                json_value.append(ftype(v))
+        else:
+            json_value = ftype(value)
+
+        json_dict[field.name] = json_value
+    return json_dict
+
+
+def _merge_json_dicts(from_dict, to_dict):
```

Comment:
Would a package work better here? Not sure about its quality though. https://pypi.org/project/jsonmerge/

Reply:
Unfortunately that package has some weird bugs: it dropped lots of fields when merging.

Reply:
Np. Just to clarify options. Custom implementation is fine.